src/PopEngine/index.ts

```diff
@@ -146,6 +151,11 @@ export class PopEngine {
         document.body.appendChild(container);
       }
 
+      if (UserAgentUtil.isSafari()) {
```

Comment:
```
 +      if (UserAgentUtil.isSafari()) {
 +        this._transitionendCallbacks[groupId] = this._removeHiddenClass.bind(this);
 +        document.querySelector('div[pop-id="' + groupId + '"]').addEventListener('transitionend', this._transitionendCallbacks[groupId], true);
 +      }
 +
```

I suspect this should be `if(UserAgentUtil.isSafari() && !this._transitionendCallbacks[groupId]) { ... }`
This way we avoid adding the multiple event listeners per pop.